mu-plugins/blocks/sidebar-container/src/view.js

```diff
@@ -26,120 +19,91 @@ let mainEl;
+	globalNavHeight = getCustomPropValue( '--wp-global-header-height' ) || 90;
+
+	containers.forEach( ( container ) => {
+		// Toggle the floating class based on the configured breakpoint.
+		const shouldFloat = window.matchMedia( `(min-width: ${ container.dataset.breakpoint })` ).matches;
+		container.classList.toggle( 'is-floating-sidebar', shouldFloat );
+		// Show the sidebar after layout, if it has been hidden to avoid FOUC.
+		if ( 'none' === window.getComputedStyle( container ).display ) {
+			container.style.setProperty( 'display', 'revert' );
+		}
+	} );
+
+	scrollHandlers.forEach( ( handler ) => handler() );
 }
 
 function init() {
-	container = document.querySelector( '.wp-block-wporg-sidebar-container' );
+	containers = document.querySelectorAll( '.wp-block-wporg-sidebar-container' );
 	mainEl = document.getElementById( 'wp--skip-link--target' );
-	const toggleButton = container?.querySelector( '.wporg-table-of-contents__toggle' );
-	const list = container?.querySelector( '.wporg-table-of-contents__list' );
-
-	if ( toggleButton && list ) {
-		toggleButton.addEventListener( 'click', function () {
-			if ( toggleButton.getAttribute( 'aria-expanded' ) === 'true' ) {
-				toggleButton.setAttribute( 'aria-expanded', false );
-				list.removeAttribute( 'style' );
-			} else {
-				toggleButton.setAttribute( 'aria-expanded', true );
-				list.setAttribute( 'style', 'display:block;' );
-			}
-		} );
-	}
-
-	if ( isSidebarWithinViewport() ) {
-		onScroll(); // Run once to avoid footer collisions on load (ex, when linked to #reply-title).
-		window.addEventListener( 'scroll', onScroll );
 
-		const observer = new window.ResizeObserver( () => {
-			// If the sidebar is positioned at the bottom and mainEl resizes,
-			// it will remain fixed at the previous bottom position, leading to a broken page layout.
-			// In this case manually trigger the scroll handler to reposition.
-			if ( container.classList.contains( 'is-bottom-sidebar' ) ) {
-				container.classList.remove( 'is-bottom-sidebar' );
-				container.style.removeProperty( 'top' );
-				const isBottom = onScroll();
-				// After the sidebar is repositioned, also adjusts the scroll position
-				// to a point where the sidebar is visible.
-				if ( isBottom ) {
-					window.scrollTo( {
-						top: container.offsetTop - FIXED_HEADER_HEIGHT,
-						behavior: 'instant',
-					} );
-				}
-			}
+	if ( mainEl && containers.length ) {
+		containers.forEach( ( container ) => {
+			const scrollHandler = createScrollHandler( container );
+			scrollHandlers.push( scrollHandler );
+			window.addEventListener( 'scroll', scrollHandler );
 		} );
-
-		observer.observe( mainEl );
 	}
 
-	// If there is no table of contents, hide the heading.
-	if ( ! document.querySelector( '.wp-block-wporg-table-of-contents' ) ) {
-		const heading = document.querySelector( '.wp-block-wporg-sidebar-container h2' );
-		heading?.style.setProperty( 'display', 'none' );
-	}
```

Comment:
Moved to the [render](https://github.com/WordPress/wporg-mu-plugins/pull/554/files#diff-df75ded5b5b15eac75d036587faef1fec21f65511f97a9fdb27c8bf87e6a7b9aR60)